client/app/src/test/java/com/example/criengine/BookUnitTest.java

```diff
@@ -0,0 +1,176 @@
+        assertEquals(newBook.getDescription(), "New Description");
+        assertEquals(newBook.getIsbn(), "New ISBN");
+        assertEquals(newBook.getStatus(), "New Status");
+        assertEquals(newBook.getBorrower(), "New Borrower");
+        assertEquals(newBook.getRequesters().size(), 0);
+        assertEquals(newBook.getGeolocation(), "New Geolocation");
+        assertEquals(newBook.getImageURL(), "New Image");
+    }
+
+    @Test
+    public void testBookConstructorThree() {
+        List<String> newRequesterList = new ArrayList<>();
+        Book newBook = new Book("New Owner", "New Title", "New Author", "New Description", "New ISBN", "New Status", "New Borrower", newRequesterList, "New Geolocation", "New Image");
+
+        assertNull(newBook.getBookID());
+        assertEquals(newBook.getOwner(), "New Owner");
+        assertEquals(newBook.getTitle(), "New Title");
+        assertEquals(newBook.getAuthor(), "New Author");
+        assertEquals(newBook.getDescription(), "New Description");
+        assertEquals(newBook.getIsbn(), "New ISBN");
+        assertEquals(newBook.getStatus(), "New Status");
+        assertEquals(newBook.getBorrower(), "New Borrower");
+        assertEquals(newBook.getRequesters().size(), 0);
+        assertEquals(newBook.getGeolocation(), "New Geolocation");
+        assertEquals(newBook.getImageURL(), "New Image");
+    }
+
+    @Test
+    public void testBookConstructorFour() {
+        List<String> newRequesterList = new ArrayList<>();
+        Book newBook = new Book("New Owner", "New Title", "New Author", "New Description", "New ISBN", "New Status");
+
+        assertNull(newBook.getBookID());
+        assertEquals(newBook.getOwner(), "New Owner");
+        assertEquals(newBook.getTitle(), "New Title");
+        assertEquals(newBook.getAuthor(), "New Author");
+        assertEquals(newBook.getDescription(), "New Description");
+        assertEquals(newBook.getIsbn(), "New ISBN");
+        assertEquals(newBook.getStatus(), "New Status");
+        assertNull(newBook.getBorrower());
+        assertEquals(newBook.getRequesters().size(), 0);
+        assertNull(newBook.getGeolocation());
+        assertNull(newBook.getImageURL());
+    }
+
+    @Test
+    public void testRequesterMethods() {
+        ArrayList<String> newRequesterList = new ArrayList<>();
+        Book newBook = new Book("ID 1", "New Owner", "New Title", "New Author", "New Description", "New ISBN", "New Status", "New Borrower", newRequesterList, "New Geolocation", "New Image");
+
+        newBook.addRequesters("A new Requester");
+        assertEquals(newBook.getRequesters().size(), 1);
+        assertEquals(newBook.getRequesters().get(0), "A new Requester");
+
+        newBook.removeRequesters("A new Requester");
+        assertEquals(newBook.getRequesters().size(), 0);
+
+        newRequesterList.add("Requester 1");
+        newRequesterList.add("Requester 2");
+        newBook.setRequesters(newRequesterList);
```

Comment:
why does a book take an arraylist of new requesters? shouldn't each user only update it once?

Reply:
I don't think this matters. The class allows for both individual additions and reassigning the entire array.

Reply:
ah sorry yea, i thought it was adding the arraylist contents to the list instead of replacing it